coremltools/converters/mil/frontend/torch/ops.py

```diff
@@ -1650,34 +1650,84 @@ def pad(context, node):
     context.add(res)
 
 
+@register_torch_op
+def adaptive_avg_pool1d(context, node):
+    _adaptive_pool1d(context, node, mb.reduce_mean)
+
+
 @register_torch_op
 def adaptive_avg_pool2d(context, node):
     _adaptive_pool2d(context, node, mb.avg_pool, mb.reduce_mean)
 
 
+def _adaptive_pool1d(context, node, reduce_op):
+    inputs = _get_inputs(context, node, expected=2)
+    x = inputs[0]
+    assert len(inputs[1].val) == 1
+    out_length = inputs[1].val[0]
+
+    if len(x.shape) == 3:
+        # 3D input
+        begin_prefix = [0, 0]
+        end_prefix = [x.shape[0], x.shape[1]]
+        out_shape = [x.shape[0], x.shape[1], out_length]
+    else:
+        # 2D input
+        assert len(x.shape) == 2
```

Comment:
would be good to add a msg saying that,
`adaptive_avg_pool1d` only support 3D/2D inputs?

Reply:
This is just a sanity check. This PyTorch op only supports 3D and 2D.